contracts/access/AccessControlDefaultAdminRules.sol

```diff
@@ -0,0 +1,255 @@
+ *   - Even after the timer has passed to avoid locking it forever.
+ * * The `DEFAULT_ADMIN_ROLE` 's admin can be only itself.
+ *
+ * Example usage:
+ *
+ * ```solidity
+ * contract MyToken is AccessControlDefaultAdminRules {
+ *   constructor() AccessControlDefaultAdminRules(
+ *     60 * 60 * 24, // 3 days
+ *     _msgSender() // Explicit initial `DEFAULT_ADMIN_ROLE` holder
+ *    ) {}
+ *}
+ * ```
+ *
+ * NOTE: The `delay` is only configurable in the constructor to avoid issues related with handling
+ * delay management during the transfer is pending to be completed.
+ *
+ * _Available since v4.9._
+ */
+abstract contract AccessControlDefaultAdminRules is IERC5313, AccessControl {
+    uint48 private immutable _delay;
+
+    address private _currentDefaultAdmin;
+    address private _pendingDefaultAdmin;
+
+    uint48 private _defaultAdminTransferDelayedUntil;
+
+    /**
+     * @dev Emitted when a `DEFAULT_ADMIN_ROLE` transfer is started, setting `newDefaultAdmin`
+     * as the next default admin, which will have rights to claim the `DEFAULT_ADMIN_ROLE`
+     * after `defaultAdminTransferDelayedUntil` is met.
+     */
+    event DefaultAdminRoleChangeStarted(address indexed newDefaultAdmin, uint48 defaultAdminTransferDelayedUntil);
+
+    /**
+     * @dev Sets the initial values for {delay} in seconds and {defaultAdmin}.
+     *
+     * The `delay` value is immutable. It can only be set at the constructor.
+     */
+    constructor(uint48 delay_, address initialDefaultAdmin) {
+        _delay = delay_;
+        _grantRole(DEFAULT_ADMIN_ROLE, initialDefaultAdmin);
+    }
+
+    /**
+     * @dev Returns the delay between each `DEFAULT_ADMIN_ROLE` transfer.
+     */
+    function delay() public view returns (uint48) {
+        return _delay;
+    }
+
+    /**
+     * @dev See {IERC5313-owner}.
+     */
+    function owner() public view virtual returns (address) {
+        return defaultAdmin();
+    }
+
+    /**
+     * @dev Returns the address of the current `DEFAULT_ADMIN_ROLE` holder.
+     */
+    function defaultAdmin() public view virtual returns (address) {
+        return _currentDefaultAdmin;
+    }
+
+    /**
+     * @dev Returns the address of the pending `DEFAULT_ADMIN_ROLE` holder.
+     */
+    function pendingDefaultAdmin() public view virtual returns (address) {
+        return _pendingDefaultAdmin;
+    }
+
+    /**
+     * @dev Returns the timestamp after which the pending default admin can claim the `DEFAULT_ADMIN_ROLE`.
+     */
+    function defaultAdminTransferDelayedUntil() public view virtual returns (uint48) {
+        return _defaultAdminTransferDelayedUntil;
+    }
+
+    /**
+     * @dev See {IERC165-supportsInterface}.
+     */
+    function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
+        return
+            interfaceId == 0x4a02b518 || // type(IAccessControlDefaultAdminRules).interfaceId
```

Comment:
Now that there is no interface anymore, we should have an comment somewhere that lists all the functions which are part of the interfaceId computation

Reply:
Do you think it's enough with the tests?
Somehow we're explicitly setting them in [these tests](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4009/files#diff-156d4e821e84d3ce615fe8bce6b2d6c21c5274752de071ab271cf84b4581d780R40-R47), but we don't pack them for release.

No strong opinion, but not sure how we'll put them in the code.

Reply:
IMO tests are not documentation. I'd have a list in the comments here.

Reply:
We agreed during a call to do the following:

- Add `IAccessControlDefaultAdmin` back to its own file
- Don't show it in the docs yet to avoid docs duplication.
- Implemented [here](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4009/commits/f420162c0eafad65187bdc219a67d5a4854beb64).